app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/components/templates.tsx

```diff
@@ -1,62 +1,137 @@
+const genBackgroundImage = () => `url(${Math.random() > 0.5 ? dataUrl1 : dataUrl2})`
 
-  return (
-    <div className="bg-white">
-      <div className="mx-auto py-16 px-4 sm:py-24 sm:px-6 lg:px-8">
-        <div className="grid grid-cols-1 gap-y-10 gap-x-6 sm:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 xl:gap-x-8">
-          {templates.map((template, index) => (
-            <div
-              key={index}
-              className="h-40 rounded-lg border-4 border-dashed border-gray-200 flex"
-            >
-              <div className="m-auto text-center">
-                <button
-                  onClick={() => {
-                    onChange(template.templateName);
-                  }}
+/** Render single template */
+interface TemplateRenderProps extends TemplatesProps {
```

Comment:
i'm not sure it makes sense for it to extend `TemplatesProps` - is there a reason you're not manually adding `onChange(name?: string): void`?

Reply:
I've changed it.
```ts
interface TemplateRenderProps {
    template: Template
    onChange(name?: string): void
}
```